plugins/woocommerce/src/Internal/Admin/Logging/Settings.php

```diff
@@ -0,0 +1,450 @@
+			'id'          => self::PREFIX . 'default_handler',
+			'type'        => 'radio',
+			'value'       => $current_value,
+			'default'     => self::DEFAULTS['default_handler'],
+			'autoload'    => false,
+			'options'     => $handler_options,
+			'disabled'    => $hardcoded ? array_keys( $handler_options ) : array(),
+			'desc'        => implode( '<br><br>', $desc ),
+			'desc_at_end' => true,
+		);
+	}
+
+	/**
+	 * The definition for the retention_period_days setting.
+	 *
+	 * @return array
+	 */
+	private function get_retention_period_days_setting_definition(): array {
+		$custom_attributes = array(
+			'min'  => 1,
+			'step' => 1,
+		);
+
+		$hardcoded = has_filter( 'woocommerce_logger_days_to_retain_logs' );
+		$desc      = '';
+		if ( $hardcoded ) {
+			$custom_attributes['disabled'] = 'true';
+
+			$desc = sprintf(
+				// translators: %s is the name of a filter hook.
+				__( 'This setting cannot be changed here because it is being set by a filter on the %s hook.', 'woocommerce' ),
+				'<code>woocommerce_logger_days_to_retain_logs</code>'
+			);
+		}
+
+		return array(
+			'title'             => __( 'Retention period', 'woocommerce' ),
+			'desc_tip'          => __( 'This sets how many days log entries will be kept before being auto-deleted.', 'woocommerce' ),
+			'id'                => self::PREFIX . 'retention_period_days',
+			'type'              => 'number',
+			'value'             => $this->get_retention_period(),
+			'default'           => self::DEFAULTS['retention_period_days'],
+			'autoload'          => false,
+			'custom_attributes' => $custom_attributes,
+			'css'               => 'width:70px;',
+			'row_class'         => 'logs-retention-period-days',
+			'suffix'            => sprintf(
+				' %s',
+				__( 'days', 'woocommerce' ),
+			),
+			'desc'              => $desc,
+		);
+	}
+
+	/**
+	 * The definition for the level_threshold setting.
+	 *
+	 * @return array
+	 */
+	private function get_level_threshold_setting_definition(): array {
+		$hardcoded = ! is_null( Constants::get_constant( 'WC_LOG_THRESHOLD' ) );
+		$desc      = '';
+		if ( $hardcoded ) {
+			$desc = sprintf(
+				// translators: %1$s is the name of a code variable. %2$s is the name of a file.
+				__( 'This setting cannot be changed here because it is defined in the %1$s constant, probably in your %2$s file.', 'woocommerce' ),
+				'<code>WC_LOG_THRESHOLD</code>',
+				'<b>wp-config.php</b>'
+			);
+		}
+
+		$labels         = WC_Log_Levels::get_all_level_labels();
+		$labels['none'] = __( 'None', 'woocommerce' );
+
+		$custom_attributes = array();
+		if ( $hardcoded ) {
+			$custom_attributes['disabled'] = 'true';
+		}
+
+		return array(
+			'title'             => __( 'Level threshold', 'woocommerce' ),
+			'desc_tip'          => __( 'This sets the minimum severity level of logs that will be processed. Lower severity levels will be ignored. "None" means all logs will be processed.', 'woocommerce' ),
```

Comment:
Should we use some other word then processed? Perhaps stored or even logged?

Reply:
Updated in cd0925da669e0b7213a7ed6e7993587fbab997ff